src/config.py

```diff
@@ -651,4 +651,35 @@ def from_network(network: Network) -> AccountingConfig:
         )
 
 
+@dataclass(frozen=True)
```

Comment:
I would move all this new code above the `class AccountingConfig`, as it is yet another part of the full configuration. Also, changes within the `AccountingConfig` class are needed to include the `OverdraftConfig`